### ISSUE TITLE
Support base path of frontend with depth > 1

### ISSUE THREAD

**Issue by bruegth** [DependencyTrack/frontend#153]
### Current Behavior:

Basepath with depth > 1 not working

When changing vue.config.js to publicPath to:
e.g. /-/dtrack/
https://github.com/DependencyTrack/frontend/blob/master/vue.config.js#L8
with NGINX config:
```
  location /-/dtrack/ {
    alias     /opt/owasp/dependency-track-frontend/;
    index     index.html;
    try_files $uri $uri/ /index.html;
  }
```

### Proposed Behavior:

Basepath with depth > 1 is working

Related source:
https://github.com/DependencyTrack/frontend/blob/master/src/views/portfolio/projects/ProjectList.vue#L84
https://github.com/DependencyTrack/frontend/blob/master/src/shared/utils.js#L67

**Comment by stevespringett** [comment]
@nscuro, correct me if I'm wrong, but didn't we remove support for non-root deployment paths of the frontend due to OIDC complexities.

I think it's expected that NGINX would have similar issues when trying to proxy as a non-root path, correct?

**Comment by nscuro** [comment]
I believe it wasn't so much because of OIDC specifically, but more so due to the split of the frontend into a separate artifact. 

`publicPath` is a built-time variable:
https://github.com/DependencyTrack/frontend/blob/508c1ef9d6295cde9608f5e4b86d30e44b4dbc0f/vue.config.js#L8

So it cannot be modified at runtime when the frontend container starts. We had the choice of leaving `publicPath` empty, which would load all resources relative to the current path, or making it absolute like it is now. The relative path was previously used, and it worked because Alpine's `WhitelistUrlFilter` did some black magic behind the scenes. So for the `bundled` distribution it was not an issue.

That being said, we can and should revisit this. Maybe there is a workaround or solution that we didn't consider the last time we looked at it.

**Comment by ngc4579** [comment]
This issue has been around for quite a while now. Is there any progress happening behind the scenes? We'd love to use Dependency Track, but are actually forced to use context paths (in contrast to a root deployment), so ATM we cannot deploy DT...

**Comment by nscuro** [comment]
In fact this was implemented by a contributor and will be shipped with v4.12: https://github.com/DependencyTrack/frontend/pull/801
